aten/src/THCUNN/ELU.cu

```diff
@@ -6,61 +6,52 @@
 template <typename T>
 struct ELUupdateOutput_functor
 {
-  const T alpha_;
+  const T poscoef_;
+  const T negcoef_;
 
-  ELUupdateOutput_functor(T alpha)
-    : alpha_(alpha)
+  ELUupdateOutput_functor(T poscoef, T negcoef)
+    : poscoef_(poscoef)
+    , negcoef_(negcoef)
   {}
 
   __device__ void operator()(T *output, const T *input) const
   {
-    *output = *input <= 0 ? (exp(*input) - 1) * alpha_ : *input;
+    *output = *input <= 0 ? (exp(*input) - 1) * poscoef_ : *input * negcoef_;
```

Comment:
what's the  definition of poscoef?  You multiply the negative inputs by the poscoef?

Reply:
oops... messed up my positives and negatives